drivers/wifi/esp_at/esp.c

```diff
@@ -825,7 +825,13 @@ static int cmd_ipd_parse_hdr(struct esp_data *dev,
 		char *remote_ip;
 		long port;
 
-		err = esp_pull_quoted(&str, str_end, &remote_ip);
+		if (!IS_ENABLED(CONFIG_WIFI_ESP_AT_VERSION_1_7)) {
+			/* Quoted per AT version 2.1.0/2.2.0 */
+			err = esp_pull_quoted(&str, str_end, &remote_ip);
+		} else {
+			/* NOT quoted per AT version 1.7.0 */
+			err = esp_pull_raw(&str, str_end, &remote_ip);
+		}
```

Comment:
```suggestion
		if (IS_ENABLED(CONFIG_WIFI_ESP_AT_VERSION_1_7)) {
			err = esp_pull_raw(&str, str_end, &remote_ip);
		} else {
			err = esp_pull_quoted(&str, str_end, &remote_ip);
		}
```

One minor change request. Please invert logic in "if" statement. It is generally easier to read:
```
if (some_statement == true) {
} else {
}
```
instead of the other way.

Reply:
Please also update commit message. Preferable there should be 2 commits for this PR:
- enable `CONFIG_WIFI_ESP_AT_CIPDINFO_USE` automatically
- handle IP unquoted parsing with ESP-AT version <2.0

Thanks!

Reply:
Updated as commented above